### ISSUE TITLE
is it possible to use an array for workspaceLayout.appsDir?

### ISSUE THREAD

**Issue by dhondiramjawade** [nrwl/nx#4905]
> is it possible to use an array for appsDir?
> 
> i'd love to do something like
> ```
> "workspaceLayout": {
>   "appsDir": ["core-use-case1/clients/**", "core-use-case2/clients/**"], 
>   "libsDir": "utils"
> }
> ```
> 
> PS: This would also mean that a a way specify the prefix `core-use-case*/clients` when adding new apps would be needed
> 
> _Originally posted by @Bulletninja in https://github.com/nrwl/nx/issues/2182#issuecomment-720232177_
> 
### @Bulletninja Suggestion is good. 
With this, we can manage 
1. Multiple clients, 
2. or differentiate our libs and application based on some categories. 

like :

- apps -> for front-end apps.
- libs   ->  for front-end libs.   
- micro-apps -> for backend micro-app.
- micro-libs   -> for backend micro-libs.

**Comment by Jordan-Hall** [comment]
I don't see the need for this. You can have a grouping folder to separate them up.

**Comment by dhondiramjawade** [comment]
Yes, @Jordan-Hall, we can group folders, but I want to group at the root level.
Because as are working on a very big project which includes 100+ libs, 
we have already categorized our libs in the following type  

- core
- feature-module
- templates 
- themes
- etc

If I import one  feature-module library then the import will be  
`import { feature } from 'workspace/feature-module/feature/' `
and if we have grouped feature then the import will be wrost like,
 `import { feature } from 'workspace/feature-module/grouped-feature/feature' `

I feel is not good to deep import with 4 or 5 levels, we can reduce one level by creating a group at the root.
If we deep import any library then the intelligence of the editor will not work correctly.
